tests/app/parser/test_supplementary_data_parser.py

```diff
@@ -0,0 +1,214 @@
+    with pytest.raises(ValidationError):
+        validate_supplementary_data_v1(
+            supplementary_data=SUPPLEMENTARY_DATA_PAYLOAD,
+            dataset_id="44f1b432-9421-49e5-bd26-e63e18a30b69",
+            unit_id="12346789012A",
+            survey_id="234",
+        )
+
+
+def test_validate_supplementary_data_payload_incorrect_unit_id():
+    with pytest.raises(ValidationError):
+        validate_supplementary_data_v1(
+            supplementary_data=SUPPLEMENTARY_DATA_PAYLOAD,
+            dataset_id="44f1b432-9421-49e5-bd26-e63e18a30b69",
+            unit_id="000000000001",
+            survey_id="123",
+        )
+
+
+def test_supplementary_data_payload_with_no_items_is_validated():
+    payload = {
+        "dataset_id": "44f1b432-9421-49e5-bd26-e63e18a30b69",
+        "survey_id": "123",
+        "data": {
+            "schema_version": "v1",
+            "identifier": "12346789012A",
+        },
+    }
+
+    validated_payload = validate_supplementary_data_v1(
+        supplementary_data=payload,
+        dataset_id="44f1b432-9421-49e5-bd26-e63e18a30b69",
+        unit_id="12346789012A",
+        survey_id="123",
+    )
+
+    assert validated_payload == payload
+
+
+def test_validate_supplementary_data_payload_missing_survey_id():
+    payload = {
+        "dataset_id": "44f1b432-9421-49e5-bd26-e63e18a30b69",
+        "data": {
+            "schema_version": "v1",
+            "identifier": "12346789012A",
+        },
+    }
+
+    with pytest.raises(ValidationError):
+        validate_supplementary_data_v1(
+            supplementary_data=payload,
+            dataset_id="44f1b432-9421-49e5-bd26-e63e18a30b69",
+            unit_id="12346789012A",
+            survey_id="123",
+        )
+
+
+def test_validate_supplementary_data_payload_with_unknown_field():
+    payload = {
+        "dataset_id": "44f1b432-9421-49e5-bd26-e63e18a30b69",
+        "survey_id": "123",
+        "some_field": "value",
+        "data": {
+            "schema_version": "v1",
+            "identifier": "12346789012A",
+        },
+    }
+
+    validated_payload = validate_supplementary_data_v1(
+        supplementary_data=payload,
+        dataset_id="44f1b432-9421-49e5-bd26-e63e18a30b69",
+        unit_id="12346789012A",
+        survey_id="123",
+    )
+
+    assert validated_payload == payload
+
+
+def test_validate_supplementary_data_invalid_schema_version():
+    payload = {
+        "dataset_id": "44f1b432-9421-49e5-bd26-e63e18a30b69",
+        "survey_id": "123",
+        "some_field": "value",
+        "data": {
+            "schema_version": "v2",
+            "identifier": "12346789012A",
+        },
+    }
+
+    with pytest.raises(ValidationError):
+        validate_supplementary_data_v1(
+            supplementary_data=payload,
+            dataset_id="001",
+            unit_id="12346789012A",
+            survey_id="123",
+        )
+
+
+def test_validate_supplementary_data_payload_missing_identifier_in_items():
+    payload = {
```

Comment:
Non blocker, could reuse SUPPLEMENTARY_DATA_PAYLOAD and pop identifier.

Current is more explicit though. But it is minor.

Reply:
I will leave this for now since it is more explicit, unless anyone has any strong views on this 👍